dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java

```diff
@@ -177,68 +185,65 @@ public void bulkUpsert(List<Entry> entries) {
-     * Gets the json used for bulk insert
-     *
-     * @param publishedEntries A list of published entries
-     * @return The json used for bulk insert
-     */
-    private String getNDJSON(List<Entry> publishedEntries) {
-        Gson gson = new GsonBuilder().create();
-        StringBuilder builder = new StringBuilder();
-        publishedEntries.forEach(entry -> {
-            entry.getWorkflowVersions().forEach(entryVersion -> {
-                ((Version)entryVersion).updateVerified();
+            @Override
+            public void afterBulk(long executionId, BulkRequest request,
+                    Throwable failure) {
+                LOGGER.error("Failed to execute bulk", failure);
+            }
+        };
+
+        try (RestHighLevelClient client = new RestHighLevelClient(
+                RestClient.builder(
+                        new HttpHost(hostname, port, "http")))) {
+            BulkProcessor.Builder builder = BulkProcessor.builder(
+                (request, bulkListener) ->
+                        client.bulkAsync(request, RequestOptions.DEFAULT, bulkListener),
+                listener);
+            // Set size of actions with `builder.setBulkSize()`, defaults to 5 MB
+            BulkProcessor bulkProcessor = builder.build();
+            entries.forEach(entry -> {
+                try {
+                    String s = MAPPER.writeValueAsString(dockstoreEntryToElasticSearchObject(entry));
+                    bulkProcessor.add(new IndexRequest(index).id(String.valueOf(entry.getId())).source(s, XContentType.JSON));
+
+                } catch (IOException e) {
+                    LOGGER.error(MAPPER_ERROR, e);
+                    throw new CustomWebApplicationException(MAPPER_ERROR, HttpStatus.SC_INTERNAL_SERVER_ERROR);
+                }
             });
-            Map<String, Map<String, String>> index = new HashMap<>();
-            Map<String, String> internal = new HashMap<>();
-            internal.put("_id", String.valueOf(entry.getId()));
-            index.put("index", internal);
-            builder.append(gson.toJson(index));
-            builder.append('\n');
             try {
-                builder.append(MAPPER.writeValueAsString(dockstoreEntryToElasticSearchObject(entry)));
-            } catch (IOException e) {
-                throw new CustomWebApplicationException(MAPPER_ERROR, HttpStatus.SC_INTERNAL_SERVER_ERROR);
+                // When doing a bulk index, this is the max amount of time the bulk listener should wait before considering the
+                // bulk request as failed. 1 minute appears to be more than enough time to index all the current Dockstore entries
+                final long bulkProcessorWaitTimeInMinutes = 1L;
+                boolean terminated = bulkProcessor.awaitClose(bulkProcessorWaitTimeInMinutes, TimeUnit.MINUTES);
+                if (!terminated) {
+                    LOGGER.error("Could not submit " + index + " index to elastic search in time");
```

Comment:
After here and all your other LOGGER.error messages I think it should throw an Exception, otherwise it will look like it worked.

Reply:
First instinct is that this would break many many tests. we'll see